### ISSUE TITLE
Proposal: require rel="noopener" on a tags

### ISSUE THREAD

**Issue by ericcornelissen** [htmllint/htmllint#232]
**tl;dr:** I want to propose the addition of an option that requires `rel="noopener"` on `<a>` tags as seen in [Lighthouse](https://developers.google.com/web/tools/lighthouse/).

* * *

## Motivation
As you can read on the Lighthouse page [about](https://developers.google.com/web/tools/lighthouse/audits/noopener) `rel="noopener"`, creating an `<a>` that has the `target="_blank"` attribute can cause both a performance slowdown (because that link is ran in the same process) and a security risk (the new page has access to your `window`).

Now can Lighthouse tell you about this issue, but it requires that the site is running. But it is possible to detect this issue statically through a linter (to a large extend, same-host links are hard to detect), which is why I'm opening this proposal.

## Details
**name:** attr-rel-noopener
**default:** disabled
**working:** There would be multiple possible ways in which this rule can work
  -  _Boolean_: The rule can be set to either `true` or `false`, so `rel="noopener"` is either always required or never. This would be the easiest to implement.
  - _Lighthouse approach_: The rule can still be set to either `true` or `false`, but instead of `rel="noopener"` always being required it is only required when there is also a `target="_blank"` on the `<a>`. Is probably the safest approach, requires little effort to understand and adhere for the user.
  - _Custom values_: The rule can accepts a custom set of values, e.g. `"always"` (so it is always required), `"lighthouse"` (to get the working as described above), `"outgoing"` (so it only applies when `target` is set, but not set to `_self`), `"none"` (so it is disabled). This provides the most freedom to the user.
  - _User defined_: The rule accepts an array of values which, if it matches the `<a>`'s value for the `target` attribute, requires `rel="noopener"`. Not sure why you would want this, just throwing out ideas 😄

**Comment by mlochbaum** [comment]
Thanks for the suggestion. `link-req-noopener` now gives an error for any tags that have `target="_blank"` but not `rel="noopener"` or `rel="noreferrer"`.

It doesn't seem to me that there's any reason to allow flexibility on the `target="_blank"` rule--that is the only case where there are performance and security issues. However, it might make sense to allow the user to require either `noopener` or `noreferrer`.

**Comment by ericcornelissen** [comment]
> Thanks for the suggestion. `link-req-noopener` now gives an error for any tags that have target="_blank" but not `rel="noopener"` or `rel="noreferrer"`.

That is awesome, thanks so much for adding this! ❤️
